src/contexts/TokenContextProvider.tsx

```diff
@@ -53,12 +43,33 @@ export function TokenContextProvider({ children }: IInit & { children: ReactNode
   const unknownTokenMap = useRef<Map<string, TokenInfo>>(new Map());
   const onChainTokenMap = useRef<Map<string, TokenInfo>>(new Map());
 
+  // Make sure initialTokenList are only fetched once
+  const [localTokenList, setLocalTokenList] = useLocalStorage<{ timestamp: number | null; data: TokenInfo[] }>(
+    'local-token-list',
+    { timestamp: null, data: [] },
+  );
+  const { data: initialTokenList } = useQuery(['cached-initial-token-list'], async () => {
+    // 10 minutes caching
+    if (localTokenList.data.length > 0 && localTokenList.timestamp && Date.now() - localTokenList.timestamp < 600_000) {
+      return localTokenList.data;
+    }
+
+    const tokens = await (await fetch('https://tokens.jup.ag/tokens?tags=strict,lst')).json();
+    const res = new TokenListContainer(tokens);
+    const list = res.getList();
+    setLocalTokenList({ timestamp: Date.now(), data: list });
+    return list;
+  });
```

Comment:
- Think we can use `staleTime` of useQuery props for auto revalidate?
- Shall we set all these to `false`? It's true by default. Or better we just set these on the single SSOT queryClient side
```
      refetchOnMount: false,
      refetchOnWindowFocus: false,
      refetchOnReconnect: false,
```

Reply:
what's SSOT?
`staleTime: true` doesn't seems to work properly on refresh, not persisted